grakn-engine/src/main/java/ai/grakn/engine/externalcomponents/CassandraSupervisor.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.externalcomponents;
+
+import ai.grakn.engine.GraknEngineConfig;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+/**
+ * Responsible for supervising cassandra.
+ *
+ * @author Ganeshwara Herawan Hananda
+ */
+
+public class CassandraSupervisor {
+    private OperatingSystemCalls osCalls;
+    private final Logger LOG = Logger.getLogger(CassandraSupervisor.class.getName());
+    private final long WAIT_TIME_BETWEEN_ATTEMPT_MS = 2000;
```

Comment:
This should probably be a configurable value to aid with debugging. 

I am just thinking, when users encounter timeout issues with grakn, one of the first things they will likely do is increase the timeout as a sanity check. I think we have all done that at some point.